src/cryptsetup/cryptsetup-generator.c

```diff
@@ -316,7 +318,7 @@ static int create_disk(
         if (keydev) {
                 _cleanup_free_ char *unit = NULL;
 
-                r = generate_keydev_mount(name, keydev, keyfile_timeout_value, keyfile_can_timeout > 0, &unit, &keydev_mount);
+                r = generate_dev_mount(name, keydev, "keydev", keyfile_timeout_value, keyfile_can_timeout > 0, true, &unit, &keydev_mount);
```

Comment:
hmm, so function that take multiple bool params become very hard to read as function calls, since you never know to what you are actually passing ht bools. please try to improve this with comments, for example:

```c
r = generate_device_mount(
        name, 
        keydev, 
        "keydev", 
        keyfile_timeout_value,
        /* canfail= */ keyfile_can_timeout > 0, 
        /* readonly= */ true,
        &unit, 
        &keydev_mount);
```

Reply:
Fixed